packages/block-library/src/navigation-link/edit.js

```diff
@@ -60,10 +60,12 @@ function NavigationLinkEdit( {
 	mergeBlocks,
 	onReplace,
 } ) {
-	const { label, opensInNewTab, url, nofollow, description } = attributes;
+	const { label, opensInNewTab, url, id, nofollow, description } = attributes;
 	const link = {
 		url,
+		id,
 		opensInNewTab,
+		title: label,
```

Comment:
Hmm, these two aren't really the same thing. If I edit the label it doesn't mean that my page title is changed.

Not sure what the alternative is, since we don't actually know the page title.